packages/keyring-controller/src/KeyringController.ts

```diff
@@ -168,6 +168,26 @@ export type KeyringControllerQRKeyringStateChangeEvent = {
   payload: [ReturnType<IQRKeyringState['getState']>];
 };
 
+export type KeyringControllerSyncIdentitiesEvent = {
+  type: `${typeof name}:syncIdentities`;
+  payload: [addresses: string[]];
+};
+
+export type KeyringControllerUpdateIdentitiesEvent = {
+  type: `${typeof name}:updateIdentities`;
+  payload: [addresses: string[]];
+};
+
+export type KeyringControllerSetSelectedAddressEvent = {
+  type: `${typeof name}:setSelectedAddress`;
+  payload: [selectedAddress: string];
+};
+
+export type KeyringControllerSetAccountLabelEvent = {
+  type: `${typeof name}:setAccountLabel`;
+  payload: [address: string, label: string];
+};
+
```

Comment:
`PreferencesController` has now responsibilities over syncing identities with the KeyringController:
https://github.com/MetaMask/core/blob/79ebb87ec9559b831e74de40efbb2d8624d4b774/packages/preferences-controller/src/PreferencesController.ts#L235

So we can simply remove all calls to `updateIdentities`, `syncIdentities`, `setSelectedAddress`, `setAccountLabel` and the corresponding constructor options from KeyringController, without adding events to the `messenger`

Reply:
Uhm, I'm not sure about `setAccountLabel` though. I don't think that's handled by PreferencesController at the moment, but I also don't feel like it should be KeyringController's responsibility to change the account label or select a different address

Reply:
Right, `setSelectedAddress` and `setAccountLabel` are now part of `AccountsController`, that's why they are not managed by `PreferencesController` (even though this is still the case on the extension). 

Given that the intention is to give this responsibility to `AccountsController`, we should keep  `setSelectedAddress` and `setAccountLabels` in place. 

We'll probably have to add some events to allow `AccountsController` to know that a label should be changed or an address should be selected - but I'm not even sure of that, as from what I can see these two methods are only used in specific cases for QRKeyring, which might go away with https://github.com/MetaMask/core/issues/3786

Reply:
Applied in https://github.com/MetaMask/core/pull/3853/commits/801026c0b66d557de43372243ba5181c0d175036, https://github.com/MetaMask/core/pull/3853/commits/806fd9b696abb1aa088bb9accce442b2852187ac!